GoogleMapsComponents/wwwroot/js/objectManager.js

```diff
@@ -390,33 +392,68 @@
             },
 
             addControls(args) {
-                let map = mapObjects[args[0]];
+                let mapGuid = args[0];
+                let map = mapObjects[mapGuid];
                 let elem = args[2];
-                //Without this the original element is removed and can not be added again.
-                let clone = elem.cloneNode(true);
+                if(!elem) return
                 //I know i am lazy. Two quotes appear after serialization
                 let position = getGooglePositionFromString(args[1].replace("\"", "").replace("\"", ""));
 
-                map.controls[position].push(clone);
+                // check if the control already exists
+                var controls = map.controls[position].getArray();
+                for (var i = 0; i < controls.length; i++) {
+                    if (controls[i].id === elem.id) {
+                        return;
+                    }
+                }
+
+                if (controlParents.hasOwnProperty(mapGuid) == false) {
+                    controlParents[mapGuid] = {}
+                }
+                if (controlParents[mapGuid].hasOwnProperty(elem.id) == false && !controlParents[mapGuid][elem.id]) {
+                    let parentElement = elem.parentElement
+                    controlParents[mapGuid][elem.id] = parentElement;
+                }
+                
+                elem.style.display = "block";
```

Comment:
Why?

Reply:
I noticed in the example code MapLegendPage that the legend originally was setting style visbility to 'hidden'. But when doing so it will still be taking up space. 
From W3Schools: 
visibility:hidden hides the element, but it still takes up space in the layout. display:none removes the element from the document.
I it to use display: none instead and then when adding the control it will set the display to "block". And when removing the control, set display to "none"